pykafka/protocol.py

```diff
@@ -137,8 +137,9 @@ class Message(Message, Serializable):
           Value => bytes
 
     :class:`pykafka.protocol.Message` also contains `partition` and
-    `partition_id` fields. Both of these have meaningless default values when
-    :class:`pykafka.protocol.Message` is used by the producer.
+    `partition_id` fields. Both of these have meaningless default values. When
+    :class:`pykafka.protocol.Message` is used by the producer, `partition_id`
+    identifies the Message's destination partition.
```

Comment:
Since you're using `Message.delivery_future` in the producer, I think it should be declared in `__init__` with `None` as its initial value. Since we're going to be keeping a lot of `Message` instances in memory now, we may want to consider adding `__slots__` to this class. `__slots__` would serve a dual purpose: explicitly documenting all of the attributes a `Message` can ever have, and reducing the memory footprint of each message instance.